check_all_options.c

```diff
@@ -101,6 +101,21 @@ void check_options_tcp_rr(struct options *opts, struct callbacks *cb)
 
 void check_options_tcp_stream(struct options *opts, struct callbacks *cb)
 {
+      if (opts->tcpd_gpu_pci_addr) {
+            CHECK(cb, opts->tcpd_nic_pci_addr,
+                  "Must provide NIC PCI address if GPU PCI address was provided.");
+            CHECK(cb, opts->tcpdirect_phys_len > 0,
+                  "Must provide non-zero --tcpdirect-phys-len flag if GPU PCI address was provided.");
+            // TODO check page-alignment
+            // CHECK((CUdeviceptr)gpu_tx_mem_ % PAGE_SIZE == 0);
+
+            if (!opts->client) {
+                  CHECK(cb, opts->tcpdirect_src_ip,
+                        "Must provide source IP address for TCPDirect host.");
+                  CHECK(cb, opts->tcpdirect_dst_ip,
+                        "Must provide destination IP address for TCPDirect host.");
```

Comment:
Need another pass over these
Off the top of my head, need rules if allocating only UDMA bufs (`else if (opts->tcpd_nic_pci_addr)`)